src/extension_commands.ml

```diff
@@ -699,23 +725,109 @@ module MerlinJump = struct
         ~preserveFocus:true
         ()
     in
-    TextEditor.set_selection
-      text_editor
-      (Selection.makePositions ~anchor:position ~active:position);
-    TextEditor.revealRange
-      text_editor
-      ~range:(Range.makePositions ~start:position ~end_:position)
-      ~revealType:TextEditorRevealType.InCenterIfOutsideViewport
-      ()
+    show_selection (Selection.makePositions ~anchor:position ~active:position) text_editor
+  ;;
+
+  module QuickPickItemWithPosition = struct
+    type t =
+      { item : QuickPickItem.t
+      ; position : Position.t
+      ; target : string
+      }
+
+    let t_of_js js =
+      let position = Ojs.get_prop_ascii js "position" |> Position.t_of_js in
+      let item = QuickPickItem.t_of_js js in
+      let target = Ojs.get_prop_ascii js "target" |> Ojs.string_of_js in
+      { item; position; target }
+    ;;
+
+    let t_to_js t =
+      let item = QuickPickItem.t_to_js t.item in
+      Ojs.set_prop_ascii item "position" (Position.t_to_js t.position);
+      Ojs.set_prop_ascii item "target" (Ojs.string_to_js t.target);
+      item
+    ;;
+  end
+
+  module QuickPick = Vscode.QuickPick.Make (QuickPickItemWithPosition)
+
+  let display_results (results : Custom_requests.Merlin_jump.response) text_editor =
+    let open Promise.Syntax in
```

Comment:
WDYT about moving the opening of `Promise.Syntax` inside the function given to the `listener` at line `804`?